src/main/java/com/igorbochkor/service/impl/CarServiceImpl.java

```diff
@@ -0,0 +1,57 @@
+package com.igorbochkor.service.impl;
+
+import com.igorbochkor.dao.CarDao;
+import com.igorbochkor.lib.Inject;
+import com.igorbochkor.lib.Service;
+import com.igorbochkor.model.Car;
+import com.igorbochkor.model.Driver;
+import com.igorbochkor.service.CarService;
+import java.util.List;
+
+@Service
+public class CarServiceImpl implements CarService {
+    @Inject
+    private CarDao carDao;
+
+    @Override
+    public Car create(Car car) {
+        return carDao.create(car);
+    }
+
+    @Override
+    public Car get(Long id) {
+        return carDao.get(id).get();
+    }
+
+    @Override
+    public List<Car> getAll() {
+        return carDao.getAllCars();
+    }
+
+    @Override
+    public Car update(Car car) {
+        return carDao.update(car);
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        return carDao.delete(id);
+    }
+
+    @Override
+    public void addDriverToCar(Driver driver, Car car) {
+        car.getDrivers().add(driver);
+        carDao.update(car);
+    }
+
+    @Override
+    public void removeDriverFromCar(Driver driver, Car car) {
+        car.getDrivers().remove(driver);
+        carDao.update(car);
+    }
```

Comment:
does it work? it seems you just add or remove from some local list, but the car data is not changing(but I'm not sure)

Reply:
sorry, even so it works

Reply:
yes, in our simulation model with a lot of assumptions it works)